### PR TITLE
Antoine feature usda api

### LINKED ISSUE #15: Stretch - Consume USDA API

### PR DESCRIPTION
**Pull request by Antoine-Aube**
## Implements/Fixes: USDA API consumption 
  
This adds an endpoint for consuming the USDA API. It parses the data to only return the necessary information to the front end. 

Potential refactors: 
- pulling out the list comprehension for data parsing into a serializer
- pulling the large method to do this API call into a service type architecture like we did in rails

NOTE: 
- This has only been tested in Postman, Will make tests separately tomorrow in separate PR! Will need to look into packages for testing APIs

## This closes issue:
 (add issue number here)
Closes #15 

## Necessary checkmarks:
- [x] All Tests are Passing
- [x] The code will run locally

## Type of change
- [x] New feature
- [ ] Bug Fix

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [x] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
      - Will test tomorrow!

TY!

### CODE REVIEW

File: market_nextdoor_api/views.py [hunk 1]
```diff
@@ -341,3 +344,35 @@ def update_preorder(preorder, data):
...
   return Response(status=status.HTTP_400_BAD_REQUEST)
+
+#USDA Market locations apikey=rTS5CqcxKA
+def get_market_locations(request, zipcode):
+  USDA_API_KEY = os.environ.get('USDA_API_KEY')
+  BASE_URL = "https://www.usdalocalfoodportal.com/api/farmersmarket/"
+  request_url = f'{BASE_URL}?apikey={USDA_API_KEY}&zip={zipcode}&radius=50'
+  head = {
+        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.85 Safari/537.36'
+    }
```
Comment (kbug819) [inline market_nextdoor_api/views.py:355]: Great find here so that we can utilize the API. Thank you for doing the extra research to resolve the issue.

**Review by EVanGorkom — APPROVED** [review]
Thank you for taking the time to figure out the python/django API call. 
Our next steps should be to include some testing, and refactor to include a services file/folder, and updating our Heroku deployment's resources with our API keys!

File: market_nextdoor_api/views.py [hunk 2]
```diff
@@ -341,3 +344,35 @@ def update_preorder(preorder, data):
...
+                'market_name': item.get('listing_name', ''),
+                'address': item.get('location_address', ''),
+                'lat': item.get('location_x', ''),
+                'lon': item.get('location_y', ''),
+                'website': item.get('media_website', ''),
+                'zipcode': item.get('location_zipcode', ''),
+                'phone': item.get('contact_phone', '')
+            }
+            for item in data.get('data', [])
+        ]
```
Comment (kbug819) [inline market_nextdoor_api/views.py:373]: Tomorrow we can look at refactoring this and possibly using the Serializer file to use SRP. Great work!

**Review by kbug819 — APPROVED** [review]

File: market_nextdoor_api/views.py (hunk 2 above)
Comment (EVanGorkom) [inline market_nextdoor_api/views.py:373]: This will may need to be refactored into a serializer.

**Pull request merged**